src/Octoshift/AdoApi.cs

```diff
@@ -425,6 +425,16 @@ private string NormalizePipelinePath(string pipeline)
             return $"\\{result}";
         }
 
+        public virtual async Task<bool> ContainsServiceConnection(string adoOrg, string adoTeamProject, string serviceConnectionId)
+        {
+            var url = $"{_adoBaseUrl}/{adoOrg}/{adoTeamProject}/_apis/serviceendpoint/endpoints/{serviceConnectionId}?api-version=6.0-preview.4";
+
+            var response = await _client.GetAsync(url);
+
+            // When the service connection isn't shared with this team project, the response is always 'null'
+            return !string.IsNullOrWhiteSpace(response) && !response.Equals("null", StringComparison.OrdinalIgnoreCase);
```

Comment:
nit: we already have an extension for `IsNullOrWhitSpace` and its negate `HasValue`, so you could re-write this as following:
```suggestion
            return response.HasValue() && !response.Equals("null", StringComparison.OrdinalIgnoreCase);
```